homeassistant/components/gdacs/geo_location.py

```diff
@@ -0,0 +1,234 @@
+"""Geolocation support for GDACS Feed."""
+import logging
+from typing import Optional
+
+from homeassistant.components.geo_location import GeolocationEvent
+from homeassistant.const import (
+    ATTR_ATTRIBUTION,
+    CONF_UNIT_SYSTEM_IMPERIAL,
+    LENGTH_KILOMETERS,
+    LENGTH_MILES,
+)
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.util.unit_system import IMPERIAL_SYSTEM
+
+from .const import (
+    DEFAULT_ICON,
+    DOMAIN,
+    FEED,
+    SIGNAL_DELETE_ENTITY,
+    SIGNAL_UPDATE_ENTITY,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_ALERT_LEVEL = "alert_level"
+ATTR_COUNTRY = "country"
+ATTR_DESCRIPTION = "description"
+ATTR_DURATION_IN_WEEK = "duration_in_week"
+ATTR_EVENT_TYPE = "event_type"
+ATTR_EXTERNAL_ID = "external_id"
+ATTR_FROM_DATE = "from_date"
+ATTR_POPULATION = "population"
+ATTR_SEVERITY = "severity"
+ATTR_TO_DATE = "to_date"
+ATTR_VULNERABILITY = "vulnerability"
+
+ICONS = {
+    "DR": "mdi:water-off",
+    "EQ": "mdi:pulse",
+    "FL": "mdi:home-flood",
+    "TC": "mdi:weather-hurricane",
+    "TS": "mdi:waves",
+    "VO": "mdi:image-filter-hdr",
+}
+
+# An update of this entity is not making a web request, but uses internal data only.
+PARALLEL_UPDATES = 100
```

Comment:
Can't we set it to 0, ie unlimited?

Reply:
Yes, unlimited makes more sense. I'll change that now.